src/main/java/transformation/TransformationP7.java

```diff
@@ -0,0 +1,160 @@
+package transformation;
+
+import common.Iterables;
+import model.*;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class TransformationP7 implements Transformation {
+
+    private static final class OrientedInterior {
+        public InteriorNode i;
+
+        public Vertex v1;
+        public Vertex v3;
+        public Vertex v5;
+
+        public Vertex h2;
+        public Vertex h4;
+
+        public GraphEdge e1;
+        public GraphEdge e2;
+        public GraphEdge e3;
+        public GraphEdge e4;
+        public GraphEdge e5;
+
+        public GraphEdge eiv1;
+        public GraphEdge eiv3;
+        public GraphEdge eiv5;
+    }
```

Comment:
Does it make sense to create additional model class? It brings inconsistency toour project, and I'm not sure if you check all posibilities. If you agree with me, use Map or something else, if not ignore and I'll merge it under your resposibility.

Reply:
This class closely matches the transformation description in the pdf exercise. AFAIS the other transformations either use Map (which I consider error-prone because when accessing graph elements by strings we can only detect access errors in runtime) or some structure made up of tuples (which I consider bonkers, because tuples completely hide semantics and instead of specific names like `foo.theVertexICareAbout` we must use generic ones `foo.getValue0` which again is error-prone; also, Java doesn't have pattern-matching, so unpacking tuples in a type-safe way is PITA).

In addition, I'd argue that the choice of internal representation is not too important when merging the code, because this representation is private to the transformation class.

Reply:
I'm not sure what you mean by checking all the possibilities.

I check whether the transformation can be applied to 3 different triangle rotations, is this what you mean?